pkg/virt-handler/vm.go

```diff
@@ -2793,7 +2793,11 @@ func (d *VirtualMachineController) vmUpdateHelperDefault(origVMI *v1.VirtualMach
 			return err
 		}
 	}
-
+	if origVMI.IsRealtimeEnabled() {
+		if err := d.configureVCPUScheduler(origVMI); err != nil {
```

Comment:
We parse the cpu mask now on every sync. Is this ok, or do we want to do book-keeping inside virt-handler memory that we did it already and try to minimize the execution? Since we are doing range operations based on masks with regexp repeatedly in an often executed path, let's double-check we are fine here.

Reply:
Done. I added a check for not running and not in final state as conditional to run the scheduling logic:
```
if vmi.IsRealtimeEnabled() && !vmi.IsRunning() && !vmi.IsFinal() {
```
I tried to move the conditional to run inside:
https://github.com/kubevirt/kubevirt/blob/ef91457727a242fb9bb742c203ac0d8866462b2f/pkg/virt-handler/vm.go#L2701
but the update never completed and ended up in an infinite loop. I suspect something is done after the conditional that makes changes to the VM state that allows this scheduling logic to work. I would have preferred to keep the new logic inside this conditional, but it didn't work out at the end.

Reply:
so does this mean that you could not find a way to make it work without calling it always? I am not sure I understand your infinite loop comment.

Reply:
Yeah sorry about that. I did find a way but let me explain: What I did was to add the condition to check if the VMI was not running and not in final phase. With that in place I saw only the logs appear only  once. I was not sure if specifying the phase to be `scheduled` or `scheduling` would be a better choice. WDYT? If you know a better way to do this please let me know and I will update it.

Forget about the infinite loop comment. When I wrote that I had just tested trying to run this logic inside vm.go:2701 and the virt-handler logs appeared to keep repeating the same messages over and over, probably because the `qemu-kvm` process was not found and it failed.